src/analyz/mod.rs

```diff
@@ -1088,7 +1126,26 @@ fn build_vtable_items<'tcx>(
     let trait_ref = tcx.instantiate_bound_regions_with_erased(trait_ref);
     let methods = tcx.vtable_entries(trait_ref);
 
-    let mut parts = Vec::with_capacity(methods.len());
+    // I'm not sure of the circumstances in which `methods` would or would not
+    // contain a `ty::vtable::VtblEntry::MetadataDropInPlace` entry, so add 1 to
+    // the allocation in case it doesn't.
+    let mut parts = Vec::with_capacity(methods.len() + 1);
+
+    // To get around an issue in which rustc seems not to generate proper drop
+    // glue for trait objects, we always include a drop method in emitted
+    // vtables (and traits - see `emit_trait` above), and we always put it at
+    // the beginning of the method list so that `crucible-mir` can find and call
+    // it manually.
+    let concrete_ty = trait_ref.self_ty();
+    let drop_inst = ty::Instance::resolve_drop_in_place(tcx, concrete_ty);
+    let drop_def_id = drop_inst.def.def_id();
+    let drop_args = drop_inst.args;
+    // Note the `.insert(0, json)` in place of `.push(json)`
+    parts.insert(0, json!({
```

Comment:
Why use `insert` here?  The vector is empty, so it's the same as a normal `push`.  I think it's fairly common to see code like
```Rust
let mut v = Vec::new();
v.push(fixed_item1);
v.push(fixed_item2);
for item in variable_items {
    v.push(f(item));
}
```